s3-commons/src/main/java/io/aiven/kafka/connect/config/s3/S3ConfigFragment.java

```diff
@@ -0,0 +1,445 @@
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_OFFSET = "offset";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_TIMESTAMP = "timestamp";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_VALUE = "value";
+    @Deprecated
+    public static final String OUTPUT_FIELD_NAME_HEADERS = "headers";
+
+    @Deprecated
+    public static final String AWS_ACCESS_KEY_ID = "aws_access_key_id";
+    @Deprecated
+    public static final String AWS_SECRET_ACCESS_KEY = "aws_secret_access_key";
+    @Deprecated
+    public static final String AWS_S3_BUCKET = "aws_s3_bucket";
+    @Deprecated
+    public static final String AWS_S3_ENDPOINT = "aws_s3_endpoint";
+    @Deprecated
+    public static final String AWS_S3_REGION = "aws_s3_region";
+    @Deprecated
+    public static final String AWS_S3_PREFIX = "aws_s3_prefix";
+    // FIXME since we support so far both old style and new style of property names
+    // Importance was set to medium,
+    // as soon we will migrate to new values it must be set to HIGH
+    // same for default value
+    public static final String AWS_ACCESS_KEY_ID_CONFIG = "aws.access.key.id";
+    public static final String AWS_SECRET_ACCESS_KEY_CONFIG = "aws.secret.access.key";
+    public static final String AWS_CREDENTIALS_PROVIDER_CONFIG = "aws.credentials.provider";
+    public static final String AWS_CREDENTIAL_PROVIDER_DEFAULT = "com.amazonaws.auth.DefaultAWSCredentialsProviderChain";
+    public static final String AWS_S3_BUCKET_NAME_CONFIG = "aws.s3.bucket.name";
+    public static final String AWS_S3_SSE_ALGORITHM_CONFIG = "aws.s3.sse.algorithm";
+    public static final String AWS_S3_ENDPOINT_CONFIG = "aws.s3.endpoint";
+    public static final String AWS_S3_REGION_CONFIG = "aws.s3.region";
+    public static final String AWS_S3_PART_SIZE = "aws.s3.part.size.bytes";
+
+    public static final String AWS_S3_PREFIX_CONFIG = "aws.s3.prefix";
+    public static final String AWS_STS_ROLE_ARN = "aws.sts.role.arn";
+    public static final String AWS_STS_ROLE_EXTERNAL_ID = "aws.sts.role.external.id";
+    public static final String AWS_STS_ROLE_SESSION_NAME = "aws.sts.role.session.name";
+    public static final String AWS_STS_ROLE_SESSION_DURATION = "aws.sts.role.session.duration";
+    public static final String AWS_STS_CONFIG_ENDPOINT = "aws.sts.config.endpoint";
+
+    public static final String AWS_S3_RETRY_BACKOFF_DELAY_MS_CONFIG = "aws.s3.backoff.delay.ms";
+    public static final String AWS_S3_RETRY_BACKOFF_MAX_DELAY_MS_CONFIG = "aws.s3.backoff.max.delay.ms";
+    public static final String AWS_S3_RETRY_BACKOFF_MAX_RETRIES_CONFIG = "aws.s3.backoff.max.retries";
+
+    private static final String GROUP_AWS = "AWS";
+    private static final String GROUP_AWS_STS = "AWS STS";
+
+    private static final String GROUP_S3_RETRY_BACKOFF_POLICY = "S3 retry backoff policy";
+
+    public static final int DEFAULT_PART_SIZE = 5 * 1024 * 1024;
```

Comment:
This config is never used. Actually this is also defined in S3OutputStream. May be reuse that or other way around.

Reply:
Yes the DEFAULT_PART_SIXE is an S3 specific piece of configuration so it has been moved to the S3ConfigFragment, once merged the S3OutputStream will be able to grab it from here as you mentioned!

Reply:
Yes, its a public static so fully available